test/test-isProbablyReaderable.js

```diff
@@ -6,18 +6,106 @@ var expect = chai.expect;
-  testPages.forEach(function(testPage) {
+describe("isProbablyReaderable - test pages", function () {
+  testPages.forEach(function (testPage) {
     var uri = "http://fakehost/test/page.html";
-    describe(testPage.dir, function() {
+    describe(testPage.dir, function () {
       var doc = new JSDOM(testPage.source, {
         url: uri,
       }).window.document;
       var expected = testPage.expectedMetadata.readerable;
-      it("The result should " + (expected ? "" : "not ") + "be readerable", function() {
+      it("The result should " + (expected ? "" : "not ") + "be readerable", function () {
         expect(isProbablyReaderable(doc)).eql(expected);
       });
     });
   });
 });
 
+describe("isProbablyReaderable", function () {
+  const makeDoc = (source) => new JSDOM(source).window.document;
+  var verySmallDoc = makeDoc("<html><p id=\"main\">hello there</p></html>"); // content length: 11
+  var smallDoc = makeDoc(`<html><p id="main">${"hello there ".repeat(11)}</p></html>`); // content length: 132
+  var largeDoc = makeDoc(`<html><p id="main">${"hello there ".repeat(12)}</p></html>`); // content length: 144
+  var veryLargeDoc = makeDoc(`<html><p id="main">${"hello there ".repeat(50)}</p></html>`); // content length: 600
+
+  it("should only declare large documents as readerable when default options", function () {
+    expect(isProbablyReaderable(verySmallDoc), "very small doc").to.be.false; // score: 0
+    expect(isProbablyReaderable(smallDoc), "small doc").to.be.false; // score: 0
+    expect(isProbablyReaderable(largeDoc), "large doc").to.be.false; // score: ~1.7
+    expect(isProbablyReaderable(veryLargeDoc), "very large doc").to.be.true; // score: ~21.4
+  });
+
+  it("should declare small and large documents as readerable when lower minContentLength", function () {
+    var options = { minContentLength: 120, minScore: 0 };
+    expect(isProbablyReaderable(verySmallDoc, options), "very small doc").to.be.false;
+    expect(isProbablyReaderable(smallDoc, options), "small doc").to.be.true;
+    expect(isProbablyReaderable(largeDoc, options), "large doc").to.be.true;
+    expect(isProbablyReaderable(veryLargeDoc, options), "very large doc").to.be.true;
+  });
+
+  it("should only declare largest document as readerable when higher minContentLength", function () {
+    var options = { minContentLength: 200, minScore: 0 };
+    expect(isProbablyReaderable(verySmallDoc, options), "very small doc").to.be.false;
+    expect(isProbablyReaderable(smallDoc, options), "small doc").to.be.false;
+    expect(isProbablyReaderable(largeDoc, options), "large doc").to.be.false;
+    expect(isProbablyReaderable(veryLargeDoc, options), "very large doc").to.be.true;
+  });
+
+  it("should declare small and large documents as readerable when lower minScore", function () {
+    var options = { minContentLength: 0, minScore: 4 };
+    expect(isProbablyReaderable(verySmallDoc, options), "very small doc").to.be.false; // score: ~3.3
```

Comment:
Tests based on score are quite fragile, but I'm not sure what would be a better idea. I would personally prefer if the function `isProbablyReaderable` would return an object with the score (and if the score is 0, a reason why), but that would be a bigger breaking change. Anyway, I think that's good enough for the current situation, but I'm open to better suggestions.

Reply:
Good idea. That would be quite weird to have an out parameter in a JS context, but that's a possible workaround. I will maybe look into this later today, I'm not sure yet if that's worth doing.